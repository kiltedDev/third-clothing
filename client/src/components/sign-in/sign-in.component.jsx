import React, { useState } from 'react';
import { connect } from 'react-redux';

import FormInput from '../form-input/form-input.component';
import CustomButton from '../custom-button/custom-button.component';

import {
  googleSignInStart,
  emailSignInStart
} from '../../redux/user/user.actions';

import {
  SignInContainer,
  SignInTitle,
  SignInButtonsContainer
} from './sign-in.styles';

const SignIn = ({ emailSignInStart, googleSignInStart }) => {
  const [ userCredentials, setUserCredentials ] = useState({
    email: '',
    password: ''
  })
  const { email, password } = userCredentials;

  const handleSubmit = async event => {
    event.preventDefault();

    emailSignInStart( email, password );
  };

  const handleChange = event => {
    const { value, name } = event.target;

    setUserCredentials({
      ...userCredentials,
      [name]: value
    });
  };

  return (
    <SignInContainer className='sign-in'>
      <SignInTitle
        children='I already have an account'
        />
      <span>Sign in with your email and password</span>
      <form onSubmit={ handleSubmit }>
        <FormInput
          name='email'
          type='email'
          handleChange={ handleChange }
          value={ email }
          label='email'

          />
        <FormInput
          name='password'
          type='password'
          value={ password }
          handleChange={ handleChange }
          label='password'

          />
        <SignInButtonsContainer className='buttons'>
          <CustomButton type='submit'> Sign in </CustomButton>
          <CustomButton
            onClick={ googleSignInStart }
            type='button'
            isGoogleSignIn
            children='Sign in with Google'
            />
        </SignInButtonsContainer>
      </form>
    </SignInContainer>
  );
}

const mapDispatchToProps = dispatch => ({
  googleSignInStart: () => dispatch( googleSignInStart() ),
  emailSignInStart: (email, password) => dispatch( emailSignInStart({ email, password }) )
})

export default connect( null, mapDispatchToProps )( SignIn );
