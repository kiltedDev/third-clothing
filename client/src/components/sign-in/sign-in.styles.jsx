import styled from 'styled-components';

export const SignInContainer = styled.div`
  width: 380px;
  max-width: 40vw;
  display: flex;
  flex-direction: column;
  button {
    max-width: 18vw;
  }
  @media screen and (max-width: 800px) {
    width: 90vw;
    button {
      min-width: unset;
    }
  }
`

export const SignInTitle = styled.h2`
  margin: 10px 0;
`

export const SignInButtonsContainer = styled.div`
  display:flex;
  justify-content: space-between;
`
