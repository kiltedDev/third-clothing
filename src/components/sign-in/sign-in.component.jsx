import React from 'react';

import FormInput from '../form-input/form-input.component';
import CustomButton from '../custom-button/custom-button.component';
import { auth } from '../../firebase/firebase.utils';


import { signInWithGoogle } from '../../firebase/firebase.utils';

import {
  SignInContainer,
  SignInTitle,
  SignInButtonsContainer
} from './sign-in.styles';

class SignIn extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: ''
    };
  }

  handleSubmit = async event => {
    event.preventDefault();

    const { email, password } = this.state;

    try {
      await auth.signInWithEmailAndPassword( email, password );
      this.setState({
        email: '',
        password: ''
      })
    } catch (e) {
      console.log(e);
    }
  };

  handleChange = event => {
    const { value, name } = event.target;

    this.setState({ [name]: value });
  };

  render() {
    return (
      <SignInContainer className='sign-in'>
        <SignInTitle
          children='I already have an account'
        />
        <span>Sign in with your email and password</span>
        <form onSubmit={this.handleSubmit}>
          <FormInput
            name='email'
            type='email'
            handleChange={this.handleChange}
            value={this.state.email}
            label='email'

          />
          <FormInput
            name='password'
            type='password'
            value={this.state.password}
            handleChange={this.handleChange}
            label='password'

          />
        <SignInButtonsContainer className='buttons'>
            <CustomButton type='submit'> Sign in </CustomButton>
            <CustomButton onClick={signInWithGoogle} isGoogleSignIn>
              Sign in with Google
            </CustomButton>
          </SignInButtonsContainer>
        </form>
      </SignInContainer>
    );
  }
}

export default SignIn;
