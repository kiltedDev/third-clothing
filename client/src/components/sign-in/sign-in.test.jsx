import { shallow } from 'enzyme';
import React from 'react';
import { SignIn } from './sign-in.component';

describe('SignIn component', () => {
  let wrapper;
  let snapshot;
  let mockGoogleSignInStart;
  let mockEmailSignInStart;
  beforeAll(() => {
    mockGoogleSignInStart = jest.fn(),
    mockEmailSignInStart = jest.fn()

    const mockProps = {
      email: 'bob@bob.net',
      password: 'iamasecurepassword',
      googleSignInStart: mockGoogleSignInStart,
      emailSignInStart: mockEmailSignInStart
    };
    wrapper = shallow(<SignIn { ...mockProps } />);
    snapshot = wrapper.debug();
  });

  it('should render SignIn component',() => {
    expect( snapshot ).toMatchSnapshot();
  });

  it('should call googleSignInStart when Google Button is clicked',() => {
    wrapper.find('CustomButton').at(1).simulate('click');
    expect(mockGoogleSignInStart).toHaveBeenCalled();
  });
});
