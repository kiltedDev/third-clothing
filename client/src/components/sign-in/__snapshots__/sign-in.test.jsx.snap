// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn component should render SignIn component 1`] = `
"<styled.div className=\\"sign-in\\">
  <styled.h2>
    I already have an account
  </styled.h2>
  <span>
    Sign in with your email and password
  </span>
  <form onSubmit={[Function: handleSubmit]}>
    <FormInput name=\\"email\\" type=\\"email\\" handleChange={[Function: handleChange]} value=\\"\\" label=\\"email\\" />
    <FormInput name=\\"password\\" type=\\"password\\" value=\\"\\" handleChange={[Function: handleChange]} label=\\"password\\" />
    <styled.div className=\\"buttons\\">
      <CustomButton type=\\"submit\\">
        Sign in
      </CustomButton>
      <CustomButton onClick={[Function: mockConstructor]} type=\\"button\\" isGoogleSignIn={true}>
        Sign in with Google
      </CustomButton>
    </styled.div>
  </form>
</styled.div>"
`;
